src/Features/CSharp/Portable/GenerateConstructor/CSharpGenerateConstructorService.cs

```diff
@@ -269,5 +269,38 @@ private static ArgumentListSyntax GetNewArgumentList(ArgumentListSyntax oldArgum
             var newArguments = oldArgumentList.Arguments.Take(argumentCount);
             return SyntaxFactory.ArgumentList(new SeparatedSyntaxList<ArgumentSyntax>().AddRange(newArguments));
         }
+
+        protected override bool CanDelegeteThisConstructor(State state, SemanticDocument document, IMethodSymbol delegatedConstructor, CancellationToken cancellationToken = default)
+        {
+            var currentConstructor = document.SemanticModel.GetDeclaredSymbol(state.Token.GetAncestor<ConstructorDeclarationSyntax>(), cancellationToken);
+            if (currentConstructor == delegatedConstructor)
+            {
+                return false;
+            }
+            else
+            {
+                while (delegatedConstructor != null)
+                {
+                    var constructorInitializerSyntax = delegatedConstructor
+                        .DeclaringSyntaxReferences[0]
+                        .GetSyntax(cancellationToken)
+                        .ChildNodes()
+                        .FirstOrDefault(n => n.IsKind(SyntaxKind.ThisConstructorInitializer)) as ConstructorInitializerSyntax;
```

Comment:
i would prefer to just cast to ConstructorDeclarationSyntax.  Then you can just grab the initializer off of that.